noah/src/main/scala/edu/uci/ics/cloudberry/noah/kafka/GeneralProducerKafka.scala

```diff
@@ -35,10 +35,13 @@ class GeneralProducerKafka(config: Config) {
     return props
   }
 
-  def store(topic: String, msg: String) {
-    val producer = new KafkaProducer[String, String](getProperties);
-    val data = new ProducerRecord[String, String](topic, msg);
-    producer.send(data);
+  def getKafkaProducer(): KafkaProducer[String, String] = {
```

Comment:
usually the `getXXX` should not create any new object. You can rename it as `createKafkaProducer` to let the caller aware of the behavior.